bin/zowe-install-component.sh

```diff
@@ -0,0 +1,182 @@
+                COMPONENT_FILE=$path
+            else
+                error_handler "-o|--component: Given path is not in a correct file format or does not exist"
+            fi
+            shift
+        ;;
+        -i|--instance_dir) #Represents the path to zowe's instance directory (optional)
+            shift
+            path=$(get_full_path "$1")
+            validate_directory_is_accessible "$path"
+            if [[ $? -eq 0 ]]; then
+                validate_file_not_in_directory "$path/instance.env" "$path"
+                if [[ $? -ne 0 ]]; then
+                    INSTANCE_DIR=$path
+                else
+                    error_handler "-i|--instance_dir: Given path is not a zowe instance directory"
+                fi
+            else
+                error_handler "-i|--instance_dir: Given path is not a zowe instance directory or does not exist"
+            fi
+            shift
+        ;;
+        -d|--target_dir) # Represents the path to the desired target directory to place the extensions (optional)
+            shift
+            TARGET_DIR=$(get_full_path "$1")
+            shift
+        ;;
+        -l|--logs-dir) # Represents the path to the installation logs
+            shift
+            LOG_DIRECTORY=$1
+            shift
+        ;;
+        *)
+            error_handler "$1 is an invalid flag\ntry: zowe-install-component.sh -o {PATH_TO_COMPONENT} -i {ZOWE_INSTANCE_DIR}"
+            shift
+    esac
+done
+
+if [ -z $COMPONENT_FILE ]; then
+    #Ensures that the required parameters are entered, otherwise exit the program
+    error_handler "Missing parameters, try: zowe-install-component.sh -e {PATH_TO_COMPONENT}"
+fi
+
+if [ -z $TARGET_DIR ]; then
+    if [ ! -z $INSTANCE_DIR ]; then #instance_dir exists
+        ZWE_EXTENSION_DIR=$(eval "grep '^ZWE_EXTENSION_DIR=' $INSTANCE_DIR/instance.env | cut -f2 -d=")
+    fi
+    if [ -z $ZWE_EXTENSION_DIR ]; then
+        #Assigns TARGET_DIR to the default directory since it was not set to a specific directory
+        TARGET_DIR=$DEFAULT_TARGET_DIR
+    else
+        TARGET_DIR=$ZWE_EXTENSION_DIR
+    fi
+fi
+
+# Checks to see if target directory is inside zowe runtime
+validate_file_not_in_directory "$TARGET_DIR" "$ZOWE_ROOT_DIR"
+if [[ $? -ne 0 ]]; then
+    error_handler "The specified target directory is located within zowe's runtime folder. Select another location for the target directory."
+fi
+
+if [ ! -e $TARGET_DIR ]; then
+    log_message "Creating extensions folder at $TARGET_DIR"
+    mkdir $TARGET_DIR
+fi
+
+if [ -z $LOG_DIRECTORY ]; then
+    LOG_DIRECTORY="$INSTANCE_DIR/logs"
+fi
+
+. ${ZOWE_ROOT_DIR}/bin/utils/setup-log-dir.sh
+set_install_log_directory "${LOG_DIRECTORY}"
+validate_log_file_not_in_root_dir "${LOG_DIRECTORY}" "${ZOWE_ROOT_DIR}"
+set_install_log_file "configure-component"
+
+log_message "Zowe Root Directory: $ZOWE_ROOT_DIR"
```

Comment:
We don't have a lint standard for shell scripts. But generally I think there are few rules we agreed on:
- capital case variables are for global purpose, it may be exported as environment variables. For local variables, we can use lower case variable names.
- `"${LOG_DIRECTORY}"` is way better in several ways:
  * curly brackets make sure the variable name will be matched exactly without any ambiguous.
  * wrap the value with `""` will make sure it works ok if there are any spaces in the value. For example, if `TARGET_DIR ="/var/my folder with space"`, the line `${ZOWE_ROOT_DIR}/bin/zowe-configure-component.sh -o $COMPONENT_NAME -d $TARGET_DIR -i $INSTANCE_DIR` will have parsing error.

Could you convert the variable name, adding proper quotations, etc.